app/src/lib/dispatcher/dispatcher.ts

```diff
@@ -816,20 +816,38 @@ export class Dispatcher {
         }
         break
 
-      case 'open-repository':
-        const { pr, url, branch } = action.args
+      case 'open-repository-from-url':
+        const { pr, url, branch } = action
         // a forked PR will provide both these values, despite the branch not existing
         // in the repository - drop the branch argument in this case so a clone will
         // checkout the default branch when it clones
         const branchToClone = (pr && branch) ? null : (branch || null)
         const repository = await this.openRepository(url, branchToClone)
         if (repository) {
-          this.handleCloneInDesktopOptions(repository, action.args)
+          this.handleCloneInDesktopOptions(repository, action)
+        }
+        break
+
+      case 'open-repository-from-path':
+        const state = this.appStore.getState()
+        const repositories = state.repositories
+        const existingRepository = repositories.find(r => (
+          Path.normalize(r.path) === Path.normalize(action.path)
```

Comment:
Does Windows always use a case-insensitive FS? On Mac, it could be either so I've tried to keep it case-preserving.